client/src/components/pages/Home/ReportBuilding/index.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { notification } from 'antd';
+import * as geocoder from 'esri-leaflet-geocoder';
+
+import Map from './Map';
+import Form from './Form';
+import styles from './report.module.css';
+
+class ReportBuilding extends Component {
+  handleCityClick = ({ latlng }) => {
+    const { handleAddressChange } = this.props;
+    const openNotificationWithIcon = (type, message) => {
+      notification[type]({
+        message,
+        duration: 3,
+      });
+    };
+
+    geocoder
+      .geocodeService()
+      .reverse()
+      .latlng(latlng)
+      .run((error, result) => {
+        if (error) {
+          handleAddressChange(undefined, undefined);
+          openNotificationWithIcon(
+            'error',
+            'Something went wrong !! Check your connection and try again',
+          );
+          return;
+        }
+        handleAddressChange(latlng, result.address.Match_addr);
```

Comment:
calling `handleAddressChange` with `undefined` for both values will cause the function to just not do anything, so why call it in the first place.

```js
        if (error)
          openNotificationWithIcon('error', 'Something went wrong!! Check your connection and try again');
        else handleAddressChange(latlng, result.address.Match_addr);
```